src/server/game/AI/CoreAI/PetAI.cpp

```diff
@@ -169,9 +169,9 @@ void PetAI::UpdateAI(uint32 diff)
                 // No enemy, check friendly
                 if (!spellUsed)
                 {
-                    for (GuidSet::const_iterator tar = _allySet.begin(); tar != _allySet.end(); ++tar)
+                    for (ObjectGuid target : _allySet)
```

Comment:
![image](https://user-images.githubusercontent.com/1153754/75098649-13ef9300-55b9-11ea-9b9d-a7f7e2693fb1.png)
I wonder what's the "standard" way to have a ObjectGuid iterator

Reply:
Well it's not really an iterator but yeah, I recon objectguid const& should be the default

Reply:
>Shauren: its definitely better to just use value for guid, not reference. at least in 3.3.5
ModoX: On master reference be better since we're using 128 bit guids over there
Shauren: @modox, yes and no. i half expect the compiler to use xmm registers like it does in wow client. in which case a copy would be better
ModoX: Good point

so copy it is :)

Reply:
Awesome, not that this holds only when not modifying the guid (and we wanna save it) 😁